### PR TITLE
MOD-6761: Add validation for cursor read `COUNT` argument

### PR DESCRIPTION
**Pull request by raz-mon**
**Describe the changes in the pull request**

This PR adds a validation for the fourth argument of the `FT.CURSOR READ` command, which is assumed to be `COUNT`.
For instance, a query like `FT.CURSOR READ idx cid LOVE 5` would be interpreted as  `FT.CURSOR READ idx cid COUNT 5`.

We also validate that the cursor subcommand is one of ['READ', 'DEL', 'GC'].

Also fixes a leak in DEBUG mode (used in sanitizer).

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/aggregate/aggregate_exec.c [hunk 1]
```diff
@@ -1152,8 +1152,19 @@ int RSCursorCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+        char err[128];
+        sprintf(err, "Unsupported argument `%s`", count_str);
+        RedisModule_ReplyWithError(ctx, err);
+        RedisModule_EndReply(reply);
+        return REDISMODULE_OK;
+      }
+
       if (RedisModule_StringToLongLong(argv[5], &count) != REDISMODULE_OK) {
         RedisModule_ReplyWithError(ctx, "Bad value for COUNT");
+        RedisModule_EndReply(reply);
```
Comment (nafraf) [inline src/aggregate/aggregate_exec.c:1167]: 1. Can we test this?
Maybe adding a test like:
`res, cid = env.cmd('FT.CURSOR', 'READ', 'idx', str(cid), 'count', '2.3')`

2. What is the expected error if `COUNT < 0` ?

**Review by nafraf — COMMENTED** [review]
Good! Some simple questions about the tests.

File: src/aggregate/aggregate_exec.c (hunk 1 above)
Comment (raz-mon) [reply src/aggregate/aggregate_exec.c:1167]: 1. Good idea. Test added.
2. 
```
127.0.0.1:6379> FT.AGGREGATE idx * withcursor count 2.3
(error) Bad arguments for COUNT: Could not convert argument to expected type
```
Test added as well

**Review by oshadmi — APPROVED** [review]

**Review by oshadmi — APPROVED** [review]

**Comment by DvirDukhan** [comment]
@raz-mon @oshadmi don't we need to backport this one?

**Comment by raz-mon** [comment]
@DvirDukhan It's breaking, so it will be added only to the next major release (as discussed with @adrianoamaral)

**Pull request merged**